soc/arm/atmel_sam/common/atmel_sam_dt.h

```diff
@@ -43,4 +45,21 @@
 		ATMEL_SAM_PIN_PERIPH(inst, idx) << 16		\
 	}
 
+/* Devicetree related macros related to clock */
+
+/** Processor Clock (HCLK) Frequency */
+#define SOC_ATMEL_SAM_HCLK_FREQ_HZ \
+		DT_PROP(DT_NODELABEL(cpu0), clock_frequency)
+
+/** Master Clock (MCK) Frequency */
+#if defined(CONFIG_SOC_ATMEL_SAME70_MDIV)
+#define SOC_ATMEL_SAM_MCK_FREQ_HZ \
+		(SOC_ATMEL_SAM_HCLK_FREQ_HZ / CONFIG_SOC_ATMEL_SAME70_MDIV)
+#elif defined(CONFIG_SOC_ATMEL_SAMV71_MDIV)
+#define SOC_ATMEL_SAM_MCK_FREQ_HZ \
+		(SOC_ATMEL_SAM_HCLK_FREQ_HZ / CONFIG_SOC_ATMEL_SAMV71_MDIV)
+#else
+#define SOC_ATMEL_SAM_MCK_FREQ_HZ SOC_ATMEL_SAM_HCLK_FREQ_HZ
+#endif /* Master Clock (MCK) Frequency */
```

Comment:
This is not really a device tree-derived value; I am not sure if it belongs in the `atmel_sam_dt.h`.

Reply:
agreed